types/wlr_surface.c

```diff
@@ -198,7 +199,8 @@ static void surface_update_damage(pixman_region32_t *buffer_damage,
 		pixman_region32_init(&surface_damage);
 		pixman_region32_copy(&surface_damage, &current->surface_damage);
 		wlr_region_transform(&surface_damage, &surface_damage,
-			current->transform, current->buffer_width, current->buffer_height);
+			wlr_output_transform_invert(current->transform),
+			current->width, current->height);
```

Comment:
Re using surface-local coordinates instead of buffer coordinates here: good catch, makes sense